lib/rspec-puppet-facts.rb

```diff
@@ -148,26 +148,16 @@ def on_supported_os_implementation(opts = {})
 
     os_facts_hash = {}
     received_facts.map do |facts|
-      # Fix facter bug
-      # Todo: refactor the whole block to rely on structured facts and use legacy facts as fallback
-      if facts[:operatingsystem] == 'Ubuntu'
-        operatingsystemmajrelease = facts[:operatingsystemrelease].split('.')[0..1].join('.')
-      elsif facts[:operatingsystem] == 'OpenBSD'
-        operatingsystemmajrelease = facts[:operatingsystemrelease]
-      elsif facts[:operatingsystem] == 'windows' && facts[:operatingsystemrelease].start_with?('10.0.')
-        operatingsystemmajrelease = '2016'
-      elsif facts.dig(:os, 'release', 'major')
-        operatingsystemmajrelease = facts[:os]['release']['major']
-      elsif facts.dig(:os, 'distro', 'release', 'major')
-        operatingsystemmajrelease = facts[:os]['distro']['release']['major']
+      if RSpec.configuration.facterdb_string_keys
+        operatingsystemmajrelease = facts['os']['release']['major']
+        operatingsystem = facts['os']['name'].downcase
+        hardwaremodel = facts['os']['hardware']
       else
-        if facts[:operatingsystemmajrelease].nil?
-          operatingsystemmajrelease = facts[:operatingsystemrelease].split('.')[0]
-        else
-          operatingsystemmajrelease = facts[:operatingsystemmajrelease]
-        end
+        operatingsystemmajrelease = facts[:os]['release']['major']
+        operatingsystem = facts[:os]['name'].downcase
+        hardwaremodel = facts[:os]['hardware']
       end
```

Comment:
How about
```ruby
os = RSpec.configuration.facterdb_string_keys ? facts[:os] : facts['os']
operatingsystemmajrelease = os['release']['major']
operatingsystem = os['name'].downcase
hardwaremodel = os['hardware']
```

Reply:
oh that's way cleaner, I will implement it.